collector/collector.go

```diff
@@ -187,9 +190,46 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
 // Collect implements the prometheus.Collector interface.
 func (c *collector) Collect(ch chan<- prometheus.Metric) {
 	wg := sync.WaitGroup{}
-	wg.Add(len(c.devices))
+
+	var realDevices []config.Device
 
 	for _, dev := range c.devices {
+		if dev.Srv != "" {
+			log.WithFields(log.Fields{
+				"SRV": dev.Srv,
+			}).Info("SRV configuration detected")
+
+			dnsMsg := new(dns.Msg)
+			dnsCli := new(dns.Client)
+
+			dnsMsg.RecursionDesired = true
+			dnsMsg.SetQuestion(dns.Fqdn(dev.Srv), dns.TypeSRV)
+			r, _, err := dnsCli.Exchange(dnsMsg, "1.1.1.1:53")
+
+			if err != nil {
+				os.Exit(1)
+			}
+
+			for _, k := range r.Answer {
+				if s, ok := k.(*dns.SRV); ok {
+					d := config.Device{}
+					d.Name = strings.TrimRight(s.Target, ".")
+					d.Address = strings.TrimRight(s.Target, ".")
+					d.User = dev.User
+					d.Password = dev.Password
+					c.getIdentity(&d)
+					realDevices = append(realDevices, d)
+				}
+			}
+		} else {
+			c.getIdentity(&dev)
```

Comment:
probably remove this and just use the device name that is in the config or from a flag on the command line. If someone has a different name on the device than in the YAML this would break that.

Reply:
I've tested it on that specific case, it is correct and doesn't break anything, what is the case where it can be wrong? Can you give me an example?

The main idea behind this is that the router identity is used mandatory, I know it is an extra request to the API before starting the scraping process but I'm currently testing with 9 different devices over VPN and it works correctlly.

I introduce this to have consistency with SRV records bacause on that we cannot use the name from the YAML and is more easy to manage when you have a large amount of devices you only need to modify the identity in your device and the metrics are updated automatically without needind to change the configuration.

Reply:
Sorry, been totally in the weeds in meat space and then the holidays happened.

For the devices that are discovered via SRV lookups I get the `getIdentity()` call before the scrape for getting a name (granted there could be some caching involved, but I digress). For devices that are read in from the config file though maybe this shouldn't be done. If the network teams label the device as something cryptic, but in the config file it's made a more human readable name (10s-swcloset-rack1) so that it's easier to grok what/where it is instead of `mt9849-10`, then the supplied information from the config file should be considered the point of truth for devices defined there, while devices discovered via the SRV lookup should rely on their information as the point of truth.

Reply:
Totally agree on your explanation and is a use case I don't identify, my last commit fixes it.